python_modules/dagster/dagster/core/executor/multiprocess.py

```diff
@@ -85,10 +85,24 @@ def execute(self):
 
 
 class MultiprocessExecutor(Executor):
-    def __init__(self, retries, max_concurrent=None):
+    def __init__(self, retries, max_concurrent, start_method):
+        # inline import to pervent accidental use of non context set multiprocessing
+        import multiprocessing
+
         self._retries = check.inst_param(retries, "retries", RetryMode)
         max_concurrent = max_concurrent if max_concurrent else multiprocessing.cpu_count()
-        self.max_concurrent = check.int_param(max_concurrent, "max_concurrent")
+        self._max_concurrent = check.int_param(max_concurrent, "max_concurrent")
+        start_method = check.str_param(start_method, "start_method")
+        valid_starts = multiprocessing.get_all_start_methods()
+        if start_method not in valid_starts:
+            raise DagsterUnmetExecutorRequirementsError(
+                f"The selected start_method '{start_method}' is not available. "
```

Comment:
maybe also print the os/python version (e.g. `sys.platform` and `sys.version`) to make our lives easier when helping users debug